xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -438,110 +472,73 @@ public void shutdown() {
 
   @VisibleForTesting
   static final class WeightedRoundRobinPicker extends SubchannelPicker {
-    private final List<ChildLbState> children;
-    private final Map<Subchannel, OrcaPerRequestReportListener> subchannelToReportListenerMap =
-        new HashMap<>();
+    private final List<ChildLbState> children; // May only be accessed from sync context
```

Comment:
I would be nice to have a note that the order must be maintained as `children`, `pickers`, `reportListeners` and `scheduler.scaledWeights` expect to have the ith element refer to the same logical item (subchannel)